tests/FakeItEasy.Specs/ResetFakeSpecs.cs

```diff
@@ -0,0 +1,398 @@
+
+        "And the method is called"
+            .x(() => result = fake.AMethod());
+
+        "Then it returns the value configured at creation time"
+            .x(() => result.Should().Be(42));
+    }
+
+    [Scenario]
+    public static void MethodConfiguredOnceAndCalledAtCreationTime(IFoo fake, int result)
+    {
+        "Given a fake with a method configured for a single call then called at creation time"
+            .x(() => fake = A.Fake<IFoo>(options =>
+            {
+                options.ConfigureFake(f =>
+                {
+                    A.CallTo(() => f.AMethod()).Returns(42).Once();
+                    f.AMethod();
+                });
+            }));
+
+        "When the fake is reset"
+            .x(() => Fake.Reset(fake));
+
+        "And the method is called"
+            .x(() => result = fake.AMethod());
+
+        "Then it returns the default value"
+            .x(() => result.Should().Be(0));
+    }
+
+    [Scenario]
+    public static void ValueTypePropertyNotSetAtCreationTime(IFoo fake, int result)
+    {
+        "Given a fake with a value-type property not set at creation time"
+            .x(() => fake = A.Fake<IFoo>());
+
+        "And the property is set"
+            .x(() => fake.AValueTypeProperty = 123);
+
+        "When the fake is reset"
+            .x(() => Fake.Reset(fake));
+
+        "And the property is called"
+            .x(() => result = fake.AValueTypeProperty);
+
+        "Then it returns the default value"
+            .x(() => result.Should().Be(0));
+    }
+
+    [Scenario]
+    public static void ValueTypePropertySetAtCreationTime(IFoo fake, int result)
+    {
+        "Given a fake with a value-type property set at creation time"
+            .x(() => fake = A.Fake<IFoo>(options =>
+            {
+                options.ConfigureFake(f => f.AValueTypeProperty = 42);
+            }));
+
+        "And the property is set differently after creation"
+            .x(() => fake.AValueTypeProperty = 123);
+
+        "When the fake is reset"
+            .x(() => Fake.Reset(fake));
+
+        "And the property is called"
+            .x(() => result = fake.AValueTypeProperty);
+
+        "Then it returns the value set at creation time"
+            .x(() => result.Should().Be(42));
+    }
+
+    [Scenario]
+    public static void ReferenceTypePropertyNotSetAtCreationTime(IFoo fake, object result)
+    {
+        "Given a fake with a reference-type property not set at creation time"
+            .x(() => fake = A.Fake<IFoo>());
+
+        "And the property is set to a given value after creation"
+            .x(() => fake.AReferenceTypeProperty = "hello");
+
+        "When the fake is reset"
+            .x(() => Fake.Reset(fake));
+
+        "And the property is called"
+            .x(() => result = fake.AReferenceTypeProperty);
+
+        "Then it returns a default value"
+            .x(() => result.Should().NotBeNull().And.NotBe("hello"));

```

Comment:
Technically all we know is that it's some random non-hello value, not that it's the default value. But what else _could_ it be? I'll allow it!

Reply:
Yeah, not sure what I was thinking. I could just test that it's an empty string

Reply:
but the type is `object`. It's not likely to be a `""`.

Reply:
Actually, no. It's an `object`, not a string. Which means I can't know what the value will be. I think the way I had done it was fine. I just need to change the step description.

Reply:
I changed the assertion to `result.Should().BeAFake()`